### PR TITLE
examples: wait_for_ready

### PR DESCRIPTION
**Pull request by canguler**

### CODE REVIEW

File: examples/features/wait_for_ready/main.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+// unaryCall makes a unary request and compares error code with the expected code.
+func unaryCall(c pb.EchoClient, requestID int, waitForReady bool, timeout time.Duration, want codes.Code) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	_, err := c.UnaryEcho(ctx, &pb.EchoRequest{Message: "Hi!"}, grpc.FailFast(!waitForReady))
+
+	got := status.Code(err)
+	if got != want {
```
[the remark this answers is not in the document]
Comment (canguler) [reply examples/features/wait_for_ready/main.go]: Should I print only if it's an error? Or, should I print in both cases (whether it is an error or not)?

[the remark this answers is not in the document]
Comment (menghanl) [reply examples/features/wait_for_ready/main.go]: I think it's OK to always print it. You can print the response when there's no error.

Also, it's probably a good idea to add a `README.md` to explain what's done and why.

File: examples/features/wait_for_ready/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Wait for ready example
+
+This example shows how to enable wait for ready (or more acurately disable fail fast) in RPC calls.
+
+This code starts the server with a 2 seconds delay. If `failfast` is not disabled, then the RPC immediately fails with `Unavailable` code (case 1). If `failfast` is disabled, then RPC waits for the server. If context dies before the server is available, then it fails with `DeadlineExceeded` (case 3). Otherwise it succeeds (case 2).
```
Comment (menghanl) [inline examples/features/wait_for_ready/README.md]: Let's use wait for ready only. Failfast is a deprecated term.

We should probably also deprecate Failfast API and add wait for ready.

File: rpc_util.go [hunk 3]
```diff
@@ -267,6 +267,11 @@ func FailFast(failFast bool) CallOption {
 	return FailFastCallOption{FailFast: failFast}
 }
 
+// WaitForReady is the oppsite of the fail fast.
+func WaitForReady(waitForReady bool) CallOption {
```
Comment (menghanl) [inline rpc_util.go]: Copy the comment of `FailFast` here.

And mark `FailFast` as deprecated with
```
// ...
// By default, RPCs are "Fail Fast".
//
// Deprecated: use WaitForReady
```

**Comment by canguler** [comment]
fixes #2490

**Pull request merged**